.github/workflows/publish-site.yml

```diff
@@ -39,8 +54,8 @@ jobs:
         id: site-changes
         if: steps.check-ghpages.outputs.exists == 'true'
         run: |
-          git diff --stat origin/gh-pages master:doc/site
-          git diff --quiet origin/gh-pages master:doc/site && (echo "modified=false" > $GITHUB_OUTPUT) || (echo "modified=true" > $GITHUB_OUTPUT)
+          git diff --stat origin/gh-pages ${{ github.ref_name }}:doc/site
+          git diff --quiet origin/gh-pages ${{ github.ref_name }}:doc/site && (echo "modified=false" > $GITHUB_OUTPUT) || (echo "modified=true" > $GITHUB_OUTPUT)
```

Comment:
Change back to master once we've tested the workflow for both site publishing and pushing to docgen repo

Reply:
I think this change is just correct. It means you can test the site generation from a branch when working on it. `github.ref_name` will be `master` when running from the main repo.

Reply:
![image](https://github.com/user-attachments/assets/3f779a8a-72ad-48f3-927e-57acf68b8d8c)

Reply:
Sure, but the problem is it will be testing _pushing to `gh_pages` branch_, which is our "production" site. We don't want any branch to publish to our official site.

Reply:
Right, but wouldn't that only be a problem if somebody chose to do build the site from a different branch on this repo? It would require a deliberate choice by one of the repo owners to do this.